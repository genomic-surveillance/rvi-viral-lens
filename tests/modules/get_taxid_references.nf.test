// Copyright (C) 2023 Genome Surveillance Unit/Genome Research Ltd.
nextflow_process {

    name "Test Process get_taxid_reference_files"
    script "modules/get_taxid_references.nf"
    process "get_taxid_reference_files"

    test("Should run without failures") {

        when {
            params {
                library_fna = "${projectDir}/tests/test_data/test_kraken_databases/minimal/library/library.fna"
                taxid = 9000003
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = params.taxid
                input[1] = params.library_fna
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
