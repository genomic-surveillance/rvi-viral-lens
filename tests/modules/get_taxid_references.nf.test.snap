{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        9000003,
                        [
                            "9000003.fa:md5,c0498775d89b9f2e5f13d00251276088",
                            "9000003.fa.amb:md5,cc8d0956a5bf7db7b7cbd0a5c163c716",
                            "9000003.fa.ann:md5,e0c6175957ae467231587ff9dd180d99",
                            "9000003.fa.bwt:md5,087da0051366c9902f9f19cb6be609e4",
                            "9000003.fa.pac:md5,4f7e8e6ceb26ed7ce135f41c019ae671",
                            "9000003.fa.sa:md5,aa6aa3a9cd5ba8e4d9e63a1497643004"
                        ],
                        ">kraken:taxid|9000003|NC_002207.1 Influenza B virus -B/Lee/1940 segment 4, complete sequence"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-24T18:01:30.36739"
    }
}